File: apps/ios/SofaExchange/Data/DTOs/ListingDTO.swift
struct ListingDTO: Decodable {
    let id: String
    let title: String
    let city: String          // raw string; mapped to domain enum in mapper
    let pricePerNightCents: Int
    let sofaType: String
    let hasFreeWifi: Bool
    let createdAt: String
    let updatedAt: String

    func toDomain() -> Listing? {
        guard let city     = City(rawValue: city),
              let sofaType = SofaType(rawValue: sofaType) else { return nil }
        return Listing(
            id: id,
            title: title,
            city: city,
            pricePerNightCents: pricePerNightCents,
            sofaType: sofaType,
            hasFreeWifi: hasFreeWifi,
            createdAt: createdAt,
            updatedAt: updatedAt
        )
    }
}

File: apps/android/app/src/main/java/com/sofaexchange/data/remote/ListingDto.kt
package com.sofaexchange.data.remote

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class ListingDto(
    @Json(name = "id")                 val id: String,
    @Json(name = "title")              val title: String,
    @Json(name = "city")               val city: String,
    @Json(name = "pricePerNightCents") val pricePerNightCents: Int,
    @Json(name = "sofaType")           val sofaType: String,
    @Json(name = "hasFreeWifi")        val hasFreeWifi: Boolean,
    @Json(name = "createdAt")          val createdAt: String,
    @Json(name = "updatedAt")          val updatedAt: String,
)
